decimal.go

```diff
@@ -767,7 +767,25 @@ func (d Decimal) RoundFloor(places int32) Decimal {
 //	NewFromFloat(1.1001).RoundUp(2).String() // output: "1.11"
 //	NewFromFloat(-1.454).RoundUp(1).String() // output: "-1.4"
 func (d Decimal) RoundUp(places int32) Decimal {
-	return newFromDecimal(d.asFallback().RoundUp(places))
+	if d.fallback != nil || places < -6 {
+		return newFromDecimal(d.asFallback().RoundUp(places))
+	}
+
+	if places >= precision {
+		// no need to round
+		return d
+	}
+
+	s := pow10Table[precision-places]
+	rescaled := d.fixed / s * s
```

Comment:
minor: we may add parenthesis in order to be more clear what is the ordering of operations?

Reply:
Thanks! Agree. Updated https://github.com/alpacahq/alpacadecimal/pull/17/commits/a30019aca7a6e94d59953f41d2af93f46a82ab28